internal/service/advancedclustertpf/resource_compatibility_reuse.go

```diff
@@ -105,22 +105,23 @@ func getAdvancedClusterContainerID(containers []admin.CloudProviderContainer, cl
 	return ""
 }
 
-func getReplicationSpecIDsFromOldAPI(ctx context.Context, projectID, clusterName string, api admin20240530.ClustersApi) (map[string]string, error) {
-	clusterOldAPI, _, err := api.GetCluster(ctx, projectID, clusterName).Execute()
+func getReplicationSpecIDsFromOldAPI(ctx context.Context, projectID, clusterName string, api admin20240530.ClustersApi) (zoneNameSpecIDs map[string]string, asymmetricShardUnsupportedError bool, err error) {
```

Comment:
[nit] `asymmetricShardUnsupportedError` -> `asymmetricShardIsUnsupported`, feel like the `Error` suffix is a bit confusing

Reply:
changed here: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2865/commits/401d84329678c7b31445e71479f852db0faa9f44